pydantic/json.py

```diff
@@ -20,7 +20,32 @@
 from .networks import NameEmail
 from .types import SecretBytes, SecretStr
 
-__all__ = 'pydantic_encoder', 'custom_pydantic_encoder', 'timedelta_isoformat'
+__all__ = (
+    'BuiltinTypeWrapper',
+    'pydantic_encoder',
+    'custom_pydantic_encoder',
+    'timedelta_isoformat',
+    'wrap_builtin_types',
+)
+
+
+class BuiltinTypeWrapper:
+    def __init__(self, obj: Any) -> None:
+        self.obj = obj
```

Comment:
I think we can have a better name than `obj`.

Like `wrapped`, `inner_obj` or similar.

Also, `__slots__`

Reply:
Fixed